### PR TITLE
Include checkoutAttemptId as a parameter in payments call.

### PR DESCRIPTION
**Pull request by nauaros**
## Summary
This PR contains work to:
- Include the `checkoutAttemptId` as part of the `PaymentComponentData` provided in the `didSubmit` method.
- Set the `checkoutAttemptId` as a parameter in the `/payments` call within session.
- Update the demo project to send the `checkoutAttemptId` in non session scenarios.

## Tested scenarios
- Unit test `checkoutAttemptId` is being within the session `/payments` request.
- Unit test `PaymentComponent` submit implementation with test class `PaymentComponentSubject`.

**Fixed issue**:  [COIOS-533](https://youtrack.is.adyen.com/issue/COIOS-533)

### CODE REVIEW

**Comment by nauaros** [comment]
Hey team! It's not possible to remove the `AdyenContext` from `AdyenSession` since it's needed to initialize the action component. Thereby, we need to keep it.

**Review by neelSharma12 — DISMISSED** [review]

File: AdyenDropIn/Components/PreApplePayComponent.swift [hunk 1]
```diff
@@ -92,7 +92,15 @@ internal final class PreApplePayComponent: PresentableComponent,
...
-        delegate?.didSubmit(data, from: self)
+        let checkoutAttemptId = component.context.analyticsProvider.checkoutAttemptId
+        let updatedData = PaymentComponentData(paymentMethodDetails: data.paymentMethod,
+                                               amount: data.amount,
+                                               order: data.order,
+                                               storePaymentMethod: data.storePaymentMethod,
+                                               browserInfo: data.browserInfo,
+                                               checkoutAttemptId: checkoutAttemptId,
+                                               installments: data.installments)
+        delegate?.didSubmit(updatedData, from: self)
```
Comment (mohammedDehairy) [inline AdyenDropIn/Components/PreApplePayComponent.swift]: this should probably use the `submit` function, right?

File: Adyen/Core/Core Protocols/PaymentComponent.swift [hunk 2]
```diff
@@ -28,13 +28,23 @@ public protocol PaymentComponent: PaymentAwareComponent, PaymentMethodAware {
...
-            delegate?.didSubmit(data, from: component ?? self)
+        let component = component ?? self
+        let checkoutAttemptId = component.context.analyticsProvider.checkoutAttemptId
+        let updatedData = PaymentComponentData(paymentMethodDetails: data.paymentMethod,
+                                               amount: data.amount,
+                                               order: data.order,
+                                               storePaymentMethod: data.storePaymentMethod,
+                                               browserInfo: data.browserInfo,
+                                               checkoutAttemptId: checkoutAttemptId,
+                                               installments: data.installments)
```
Comment (mohammedDehairy) [inline Adyen/Core/Core Protocols/PaymentComponent.swift:44]: here you can probably check if `data.checkoutAttemptId` is already equal to `component.context.analyticsProvider.checkoutAttemptId` before updating.

File: AdyenDropIn/DropInComponentExtensions.swift [hunk 3]
```diff
@@ -51,7 +51,17 @@ extension DropInComponent: PaymentComponentDelegate {
...
+        let checkoutAttemptId = component.context.analyticsProvider.checkoutAttemptId
+        let updatedData = PaymentComponentData(paymentMethodDetails: data.paymentMethod,
+                                               amount: data.amount,
+                                               order: data.order,
+                                               storePaymentMethod: data.storePaymentMethod,
+                                               browserInfo: data.browserInfo,
+                                               checkoutAttemptId: checkoutAttemptId,
+                                               installments: data.installments)
+
+        delegate?.didSubmit(updatedData, from: component, in: self)
```
Comment (mohammedDehairy) [inline AdyenDropIn/DropInComponentExtensions.swift:70]: this should probably use the `submit` function, right?

Comment (nauaros) [reply AdyenDropIn/DropInComponentExtensions.swift:70]: Hey! `DropInComponent` doesn't conform to `PaymentComponent`. That's the reason why we're not using the `submit` function here.

Comment (mohammedDehairy) [reply AdyenDropIn/DropInComponentExtensions.swift:70]: right, sorry :-D

File: AdyenDropIn/Components/PreApplePayComponent.swift (hunk 1 above)
Comment (nauaros) [reply AdyenDropIn/Components/PreApplePayComponent.swift]: Yes, you're right! Let me update it

Comment (nauaros) [reply AdyenDropIn/Components/PreApplePayComponent.swift]: Done!

File: Adyen/Core/Core Protocols/PaymentComponent.swift (hunk 2 above)
Comment (nauaros) [reply Adyen/Core/Core Protocols/PaymentComponent.swift:44]: Updated

File: AdyenDropIn/Components/PreApplePayComponent.swift [hunk 4]
```diff
@@ -92,7 +92,22 @@ internal final class PreApplePayComponent: PresentableComponent,
...
+            updatedData = data
+        } else {
+            updatedData = PaymentComponentData(paymentMethodDetails: data.paymentMethod,
+                                               amount: data.amount,
+                                               order: data.order,
+                                               storePaymentMethod: data.storePaymentMethod,
+                                               browserInfo: data.browserInfo,
+                                               checkoutAttemptId: checkoutAttemptId,
+                                               installments: data.installments)
+        }
```
Comment (mohammedDehairy) [inline AdyenDropIn/Components/PreApplePayComponent.swift]: I don't think this is needed, right? the `submit(data: updatedData, component: self)` already does this, right?

Comment (nauaros) [reply AdyenDropIn/Components/PreApplePayComponent.swift]: Fixed

**Review by mohammedDehairy — DISMISSED** [review]

**Review by neelSharma12 — DISMISSED** [review]

**Review by neelSharma12 — DISMISSED** [review]

**Review by erenbesel — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

**Pull request merged**